apps/passport-client/components/screens/LoginScreens/LoginScreen.tsx

```diff
@@ -126,6 +130,78 @@ export function LoginScreen(): JSX.Element {
     [dispatch, email]
   );
 
+  const [tryStorageAccess, setTryStorageAccess] = useState(false);
+  const [tryStorageLogin, setTryStorageLogin] = useState(false);
+
+  const tryToLogin = useCallback(
+    async (encryptionKey: string) => {
+      try {
+        // Try to download and decrypt the storage
+        const storageRequest = await requestDownloadAndDecryptStorage(
+          appConfig.zupassServer,
+          encryptionKey
+        );
+        if (storageRequest.success) {
+          // Success, log in
+          dispatch({
+            type: "load-after-login",
+            storage: storageRequest.value,
+            encryptionKey
+          });
+        } else {
+          // Failed to log in with key from local storage
+          // This will cause the regular login flow to kick in
+          setTryStorageLogin(false);
+        }
+      } catch (e) {
+        setTryStorageLogin(false);
+      }
+    },
+    [dispatch]
+  );
+
+  const requestStorageAccessAndLogIn = useCallback(() => {
+    const parser = new UAParser();
+    const isChrome = parser.getBrowser().name === "Chrome";
+    if (!isChrome) {
+      setTryStorageAccess(false);
+      return;
+    }
+
+    document
+      // @ts-expect-error Chrome-only API
+      .requestStorageAccess({ localStorage: true })
+      // @ts-expect-error Chrome-only API
+      .then((handle: { localStorage: Storage }) => {
+        // Access granted, try reading the local storage
+        const encryptionKey = handle.localStorage.getItem("encryption_key");
+        if (encryptionKey) {
+          return tryToLogin(encryptionKey);
+        }
+      })
+      .catch(() => {})
```

Comment:
Is this line what's expected to handle the case of non-Chrome browsers?  What does the user see in this case?  Normal login screen in the iframe?

Reply:
If we're in a non-Chrome browser then this request will not have happened - all non-Chrome browsers return on line 168.

This catches the exception thrown if the user rejects the storage request, or has some browser setting or extension which causes it to be rejected. By calling `setTryStorageAccess(false)` we turn off the display of the storage access request and return the user to the regular login flow.